gui/gui/src/TGSpeedo.cxx

```diff
@@ -387,6 +413,16 @@ void TGSpeedo::SetScaleValue(Float_t val)
    if (fValue > fPeakVal)
       fPeakVal = fValue;
 
+   if (fBufferSize > 0) {
+      if ((Int_t)fBuffer.size() < (fBufferCount + 1))
+         fBuffer.push_back(fValue);
+      else
+         fBuffer[fBufferCount % fBufferSize] = fValue;
```

Comment:
Because of line 422, don't we always have `(fBufferCount % fBufferSize) == fBufferCount` ?

Reply:
Yes, you're right

Reply:
As already mentioned, that was an extra protection in the original code...

Reply:
I agree with @pcanal , iiuc the contract is that `0 <= fBufferCount < fBufferSize`